src/main/java/duke/tasks/Deadline.java

```diff
@@ -0,0 +1,46 @@
+package duke.tasks;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
+/**
+ * Deadline is a class which represents the deadline task.
+ */
+public class Deadline extends Task{
+    protected LocalDate by;
```

Comment:
Perhaps the variable name here could be more descriptive? Maybe deadlineDate would describe it better?